goosebit/auth/__init__.py

```diff
@@ -117,15 +117,19 @@ def validate_user_permissions(
 
 
 def compare_permissions(scopes: list[str] | None, permissions: set[str]) -> bool:
+    deny_permissions = [p.replace("!", "") for p in permissions if p.startswith("!")]
```

Comment:
Could be a good area to write a few unit tests (like https://github.com/UpstreamDataInc/goosebit/blob/master/tests/updates/test_swdesc.py). This could encode the description that comes with this PR and guarantee that it also works.

Reply:
Good call, I'll see what I can do here.

Reply:
Attempted this, let me know how they are.  Not very good at writing tests, but I think these should adequately test the functionality.